model/Hierarchy.php

```diff
@@ -238,8 +247,9 @@ public function setMarkingFilterFunction($funcName) {
 
 	/**
 	 * Returns true if the marking filter matches on the given node.
-	 * @param DataObject $node Node to check.
-	 * @return boolean
+	 *
+	 * @param DataObject $node Node to check
```

Comment:
I was worried that changing type-hinting could (in some cases) break existing code which would require a major version increase, whereas only updating documentation wouldn't have this risk. So I made the decision to make this PR documentation-only to assist approval and leave the code-changes to smaller specific PRs.